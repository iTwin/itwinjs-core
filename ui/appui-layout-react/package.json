{
  "name": "@itwin/appui-layout-react",
  "version": "3.3.0-dev.11",
  "description": "iTwin.js Nine-zone React UI components",
  "main": "lib/cjs/appui-layout-react.js",
  "module": "lib/esm/appui-layout-react.js",
  "typings": "lib/cjs/appui-layout-react",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/iTwin/itwinjs-core/tree/master/ui/appui-layout-react"
  },
  "scripts": {
    "build": "npm run -s build:cjs",
    "copy:cjs": "cpx \"./src/**/*.{*css,svg}\" \"./lib/cjs\"",
    "copy:esm": "cpx \"./src/**/*.{*css,svg}\" \"./lib/esm\"",
    "build:ci": "npm run -s build && npm run -s build:esm",
    "build:cjs": "tsc 1>&2 --outDir lib/cjs && npm run -s copy:cjs",
    "build:esm": "tsc 1>&2 --module ES2020 --outDir lib/esm && npm run -s copy:esm",
    "clean": "rimraf lib .rush/temp/package-deps*.json",
    "cover": "nyc npm -s test",
    "docs": "npm run -s docs:typedoc",
    "docs:typedoc": "betools docs --includes=../../generated-docs/extract --json=../../generated-docs/ui/appui-layout-react/file.json --excludeGlob=**/declarations.d.ts --tsIndexFile=./appui-layout-react.ts --onlyJson",
    "extract-api": "betools extract-api --entry=appui-layout-react",
    "lint": "eslint -f visualstudio \"./src/**/*.{ts,tsx}\" 1>&2",
    "test": "mocha --config ../.mocharc.json \"./lib/cjs/test/**/*.test.js\"",
    "test:watch": "npm -s test -- --reporter min --watch-extensions ts,tsx --watch"
  },
  "keywords": [
    "Bentley",
    "BIM",
    "iModel"
  ],
  "author": {
    "name": "Bentley Systems, Inc.",
    "url": "http://www.bentley.com"
  },
  "peerDependencies": {
    "@itwin/appui-abstract": "workspace:^3.3.0-dev.11",
    "@itwin/core-bentley": "workspace:^3.3.0-dev.11",
    "@itwin/core-react": "workspace:^3.3.0-dev.11",
    "react": "^17.0.0",
    "react-dom": "^17.0.0"
  },
  "//devDependencies": [
    "NOTE: All peerDependencies should also be listed as devDependencies since peerDependencies are not considered by npm install",
    "NOTE: All tools used by scripts in this package must be listed as devDependencies"
  ],
  "devDependencies": {
    "@itwin/appui-abstract": "workspace:*",
    "@itwin/build-tools": "workspace:*",
    "@itwin/core-bentley": "workspace:*",
    "@itwin/core-react": "workspace:*",
    "@itwin/eslint-plugin": "workspace:*",
    "@testing-library/react": "^12.0.0",
    "@testing-library/react-hooks": "^7.0.2",
    "@types/chai": "^4.1.4",
    "@types/chai-as-promised": "^7",
    "@types/chai-jest-snapshot": "^1.3.0",
    "@types/chai-spies": "^1.0.0",
    "@types/enzyme": "3.9.3",
    "@types/mocha": "^8.2.2",
    "@types/node": "16.11.7",
    "@types/react": "^17.0.37",
    "@types/react-dom": "^17.0.0",
    "@types/react-transition-group": "^4.4.4",
    "@types/sinon": "^9.0.0",
    "@types/uuid": "^7.0.2",
    "chai": "^4.1.2",
    "chai-as-promised": "^7",
    "chai-jest-snapshot": "^2.0.0",
    "chai-spies": "1.0.0",
    "cpx2": "^3.0.0",
    "enzyme": "^3.4.0",
    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.3",
    "enzyme-to-json": "^3.3.4",
    "eslint": "^7.11.0",
    "ignore-styles": "^5.0.1",
    "jsdom": "^19.0.0",
    "jsdom-global": "3.0.2",
    "mocha": "^10.0.0",
    "nyc": "^15.1.0",
    "raf": "^3.4.0",
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "rimraf": "^3.0.2",
    "sinon": "^9.0.2",
    "sinon-chai": "^3.2.0",
    "ts-node": "^10.4.0",
    "typemoq": "^2.1.0",
    "typescript": "~4.4.0",
    "xmlhttprequest": "^1.8.0"
  },
  "//dependencies": [
    "NOTE: these dependencies should be only for things that DO NOT APPEAR IN THE API",
    "NOTE: core-frontend should remain UI technology agnostic, so no react/angular dependencies are allowed"
  ],
  "dependencies": {
    "@itwin/itwinui-css": "0.x",
    "@itwin/itwinui-react": "^1.32.0",
    "classnames": "^2.3.1",
    "immer": "9.0.6",
    "react-transition-group": "^4.4.2",
    "uuid": "^7.0.3"
  },
  "nyc": {
    "extends": "./node_modules/@itwin/build-tools/.nycrc",
    "require": [
      "ignore-styles",
      "jsdom-global/register",
      "source-map-support/register",
      "ts-node/register"
    ],
    "check-coverage": true,
    "statements": 100,
    "branches": 100,
    "functions": 100,
    "lines": 100
  },
  "eslintConfig": {
    "plugins": [
      "@itwin",
      "jsx-a11y"
    ],
    "extends": [
      "plugin:@itwin/ui",
      "plugin:@itwin/jsdoc"
    ],
    "rules": {
      "deprecation/deprecation": "off"
    }
  }
}
