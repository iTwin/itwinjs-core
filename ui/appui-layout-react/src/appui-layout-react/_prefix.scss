/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@mixin nz-backface-visibility($value) {
  -webkit-backface-visibility: $value;
  -moz-backface-visibility: $value;
  backface-visibility: $value;
}

@mixin nz-transition($value) {
  -webkit-transition: $value;
  -moz-transition: $value;
  -o-transition: $value;
  transition: $value;
}

@mixin nz-transform($value) {
  -moz-transform: $value;
  -webkit-transform: $value;
  transform: $value;
}

@mixin nz-border-radius($value) {
  -webkit-border-radius: $value;
  -moz-border-radius: $value;
  border-radius: $value;
}

@mixin nz-cursor-grabbing() {
  cursor: -webkit-grabbing;
  cursor: grabbing;
}
