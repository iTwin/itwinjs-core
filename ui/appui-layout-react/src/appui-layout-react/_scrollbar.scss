/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "~@itwin/core-react/lib/cjs/core-react/scrollbar";

$nz-scrollbar-width: 12px;
$nz-scrollbar-padding: 2px;

@mixin nz-scrollbar($color: $buic-background-scrollbar, $width: $nz-scrollbar-width, $padding: $nz-scrollbar-padding) {
  @include uicore-scrollbar($color, $width, $padding);
}

@mixin nz-hidden-scrollbar {
  @include uicore-hidden-scrollbar;
}