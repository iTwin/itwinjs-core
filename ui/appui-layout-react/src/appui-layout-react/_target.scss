/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@mixin nz-internal-targeted {
  &:hover, &.nz-targeted {
    @content;
  }
}
