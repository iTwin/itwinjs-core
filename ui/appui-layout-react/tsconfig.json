{
  "extends": "./node_modules/@itwin/build-tools/tsconfig-base.json",
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "jsx": "react",
    "types": ["chai-as-promised", "mocha", "chai-jest-snapshot"]
  },
  "include": ["./src/**/*.ts*"]
}
