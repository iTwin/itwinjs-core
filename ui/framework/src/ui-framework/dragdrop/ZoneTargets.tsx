/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Zone
 */

import * as React from "react";
import { CommonProps } from "@bentley/ui-core";
import { BackTarget, MergeTarget, WidgetZoneId, ZoneTargetType } from "@bentley/ui-ninezone";
import { TargetChangeHandler } from "../frontstage/FrontstageComposer";

/** Properties for the [[ZoneTargets]] component.
 * @internal
 */
export interface ZoneTargetsProps extends CommonProps {
  zoneId: WidgetZoneId;
  dropTarget: ZoneTargetType | undefined;
  targetChangeHandler: TargetChangeHandler;
}

/** Zone Targets React component.
 * @internal
 */
export class ZoneTargets extends React.Component<ZoneTargetsProps> {
  public override render(): React.ReactNode {
    switch (this.props.dropTarget) {
      case ZoneTargetType.Merge:
        return (
          <MergeTarget
            className={this.props.className}
            onTargetChanged={(isTargeted) => this.props.targetChangeHandler.handleTargetChanged(this.props.zoneId, ZoneTargetType.Merge, isTargeted)}
            style={this.props.style}
          />
        );
      case ZoneTargetType.Back:
        return (
          <BackTarget
            className={this.props.className}
            onTargetChanged={(isTargeted) => this.props.targetChangeHandler.handleTargetChanged(this.props.zoneId, ZoneTargetType.Back, isTargeted)}
            style={this.props.style}
            zoneIndex={this.props.zoneId}
          />
        );
      default:
        return null;
    }
  }
}
