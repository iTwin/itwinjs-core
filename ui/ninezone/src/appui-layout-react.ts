/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
export * from "./appui-layout-react/backstage/Backstage";
export * from "./appui-layout-react/backstage/Item";
export * from "./appui-layout-react/backstage/Separator";
export * from "./appui-layout-react/backstage/UserProfile";

export * from "./appui-layout-react/base/DragHandle";
export * from "./appui-layout-react/base/DragManager";
export * from "./appui-layout-react/base/Ellipsis";
export * from "./appui-layout-react/base/Event";
export * from "./appui-layout-react/base/NineZone";
export * from "./appui-layout-react/base/NineZoneState";
export * from "./appui-layout-react/base/PointerCaptor";
export * from "./appui-layout-react/base/WithContainIn";

export * from "./appui-layout-react/footer/dialog/Button";
export * from "./appui-layout-react/footer/dialog/Dialog";
export * from "./appui-layout-react/footer/dialog/TitleBar";

export * from "./appui-layout-react/footer/message/Button";
export * from "./appui-layout-react/footer/message/Hyperlink";
export * from "./appui-layout-react/footer/message/Layout";
export * from "./appui-layout-react/footer/message/Message";
export * from "./appui-layout-react/footer/message/Progress";
export * from "./appui-layout-react/footer/message/Status";
export * from "./appui-layout-react/footer/message/Toast";

export * from "./appui-layout-react/footer/message-center/Dialog";
export * from "./appui-layout-react/footer/message-center/Indicator";
export * from "./appui-layout-react/footer/message-center/Message";
export * from "./appui-layout-react/footer/message-center/Tab";

export * from "./appui-layout-react/footer/snap-mode/Indicator";
export * from "./appui-layout-react/footer/snap-mode/Panel";
export * from "./appui-layout-react/footer/snap-mode/Snap";

export * from "./appui-layout-react/footer/tool-assistance/Dialog";
export * from "./appui-layout-react/footer/tool-assistance/Indicator";
export * from "./appui-layout-react/footer/tool-assistance/Instruction";
export * from "./appui-layout-react/footer/tool-assistance/Item";
export * from "./appui-layout-react/footer/tool-assistance/Separator";

export * from "./appui-layout-react/footer/Footer";
export * from "./appui-layout-react/footer/Indicator";
export * from "./appui-layout-react/footer/Popup";
export * from "./appui-layout-react/footer/Separator";

export * from "./appui-layout-react/manager/NestedStagePanels";
export * from "./appui-layout-react/manager/NineZone";
export * from "./appui-layout-react/manager/StagePanel";
export * from "./appui-layout-react/manager/StagePanelPane";
export * from "./appui-layout-react/manager/StagePanels";

export * from "./appui-layout-react/popup/Tooltip";

export * from "./appui-layout-react/stage-panels/Splitter";
export * from "./appui-layout-react/stage-panels/StagePanel";
export * from "./appui-layout-react/stage-panels/StagePanels";

export * from "./appui-layout-react/stage-panels/manager/NestedStagePanels";
export * from "./appui-layout-react/stage-panels/manager/StagePanel";
export * from "./appui-layout-react/stage-panels/manager/StagePanels";

export * from "./appui-layout-react/tool-settings/Docked";
export * from "./appui-layout-react/tool-settings/Handle";
export * from "./appui-layout-react/tool-settings/Overflow";
export * from "./appui-layout-react/tool-settings/Panel";
export * from "./appui-layout-react/tool-settings/Setting";

export * from "./appui-layout-react/toolbar/item/Item";
export * from "./appui-layout-react/toolbar/item/Overflow";
export * from "./appui-layout-react/toolbar/item/expandable/Expandable";
export * from "./appui-layout-react/toolbar/item/expandable/WithDragInteraction";

export * from "./appui-layout-react/toolbar/item/expandable/group/BackArrow";
export * from "./appui-layout-react/toolbar/item/expandable/group/Column";
export * from "./appui-layout-react/toolbar/item/expandable/group/Columns";
export * from "./appui-layout-react/toolbar/item/expandable/group/Group";
export * from "./appui-layout-react/toolbar/item/expandable/group/Nested";
export * from "./appui-layout-react/toolbar/item/expandable/group/Panel";
export * from "./appui-layout-react/toolbar/item/expandable/group/Title";

export * from "./appui-layout-react/toolbar/item/expandable/group/tool/Expander";
export * from "./appui-layout-react/toolbar/item/expandable/group/tool/Tool";

export * from "./appui-layout-react/toolbar/Items";
export * from "./appui-layout-react/toolbar/Toolbar";

export * from "./appui-layout-react/utilities/Cell";
export * from "./appui-layout-react/utilities/Css";
export * from "./appui-layout-react/utilities/Direction";
export * from "./appui-layout-react/utilities/DisabledResizeHandles";
export * from "./appui-layout-react/utilities/SafeAreaInsets";

export * from "./appui-layout-react/widget/Buttons";
export * from "./appui-layout-react/widget/Content";
export * from "./appui-layout-react/widget/ContentContainer";
export * from "./appui-layout-react/widget/ContentManager";
export * from "./appui-layout-react/widget/ContentRenderer";
export * from "./appui-layout-react/widget/Dock";
export * from "./appui-layout-react/widget/FloatingTab";
export * from "./appui-layout-react/widget/FloatingWidget";
export * from "./appui-layout-react/widget/FloatingWidgets";
export * from "./appui-layout-react/widget/Menu";
export * from "./appui-layout-react/widget/NavigationArea";
export * from "./appui-layout-react/widget/Overflow";
export * from "./appui-layout-react/widget/PanelWidget";
export * from "./appui-layout-react/widget/PinToggle";
export * from "./appui-layout-react/widget/PopoutToggle";
export * from "./appui-layout-react/widget/SendBack";
export * from "./appui-layout-react/widget/Stacked";
export * from "./appui-layout-react/widget/Tab";
export * from "./appui-layout-react/widget/TabBar";
export * from "./appui-layout-react/widget/Tabs";
export * from "./appui-layout-react/widget/TabTarget";
export * from "./appui-layout-react/widget/Tools";
export * from "./appui-layout-react/widget/ToolsArea";
export * from "./appui-layout-react/widget/ToolSettings";
export * from "./appui-layout-react/widget/Widget";
export * from "./appui-layout-react/widget/WidgetTarget";

export * from "./appui-layout-react/widget/tools/button/App";
export * from "./appui-layout-react/widget/tools/button/Back";
export * from "./appui-layout-react/widget/tools/button/Button";
export * from "./appui-layout-react/widget/tools/button/Expandable";
export * from "./appui-layout-react/widget/tools/button/Icon";

export * from "./appui-layout-react/widget/rectangular/Content";
export * from "./appui-layout-react/widget/rectangular/ResizeGrip";

export * from "./appui-layout-react/widget/rectangular/tab/Group";
export * from "./appui-layout-react/widget/rectangular/tab/Separator";
export * from "./appui-layout-react/widget/rectangular/tab/Tab";

export * from "./appui-layout-react/widget/tool-settings/Tab";

export * from "./appui-layout-react/widget-panels/AppContent";
export * from "./appui-layout-react/widget-panels/CenterContent";
export * from "./appui-layout-react/widget-panels/Content";
export * from "./appui-layout-react/widget-panels/CursorOverlay";
export * from "./appui-layout-react/widget-panels/Expander";
export * from "./appui-layout-react/widget-panels/Grip";
export * from "./appui-layout-react/widget-panels/Panel";
export * from "./appui-layout-react/widget-panels/Panels";
export * from "./appui-layout-react/widget-panels/PanelTarget";

export * from "./appui-layout-react/zones/Outline";
export * from "./appui-layout-react/zones/Zone";
export * from "./appui-layout-react/zones/Zones";

export * from "./appui-layout-react/zones/manager/ResizeStrategy";
export * from "./appui-layout-react/zones/manager/Widget";
export * from "./appui-layout-react/zones/manager/Zone";
export * from "./appui-layout-react/zones/manager/Zones";

export * from "./appui-layout-react/zones/target/Back";
export * from "./appui-layout-react/zones/target/Merge";
export * from "./appui-layout-react/zones/target/Splitter";
export * from "./appui-layout-react/zones/target/SplitterPane";
export * from "./appui-layout-react/zones/target/StagePanel";

/** @docs-package-description
 * The appui-layout-react package contains React components for application user interface layouts following the Bentley 9-Zone pattern.
 * For more information, see [learning about appui-layout-react]($docs/learning/ui/ninezone/index.md).
 */
/**
 * @docs-group-description Backstage
 * Classes for working with a Backstage
 */
/**
 * @docs-group-description Footer
 * Classes for working with a Footer or Status Bar
 */
/**
 * @docs-group-description Message
 * Classes for working with Messages
 */
/**
 * @docs-group-description MessageCenter
 * Classes for working with the MessageCenter
 */
/**
 * @docs-group-description Popup
 * Classes for working with a Popup
 */
/**
 * @docs-group-description SnapMode
 * Classes for working with SnapMode UI
 */
/**
 * @docs-group-description StagePanels
 * Classes for working with stage panels
 */
/**
 * @docs-group-description ToolAssistance
 * Classes for working with Tool Assistance
 */
/**
 * @docs-group-description Toolbar
 * Classes for working with a Toolbar
 */
/**
 * @docs-group-description ToolSettings
 * Classes for working with Tool Settings
 */
/**
 * @docs-group-description Utilities
 * Utility classes for working with the 9-Zone pattern
 */
/**
 * @docs-group-description Widget
 * Classes for working a Widget
 */
/**
 * @docs-group-description WidgetPanels
 * Classes for working with widget panels
 */
/**
 * @docs-group-description Zone
 * Classes for working a Zone
 */
