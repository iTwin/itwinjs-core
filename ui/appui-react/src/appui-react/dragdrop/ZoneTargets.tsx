/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Zone
 */

import * as React from "react";
import { CommonProps } from "@itwin/core-react";
import { BackTarget, MergeTarget, WidgetZoneId, ZoneTargetType } from "@itwin/appui-layout-react";
import { TargetChangeHandler } from "../frontstage/FrontstageComposer";

/** Properties for the [[ZoneTargets]] component.
 * @internal
 */
export interface ZoneTargetsProps extends CommonProps {
  zoneId: WidgetZoneId; // eslint-disable-line deprecation/deprecation
  dropTarget: ZoneTargetType | undefined; // eslint-disable-line deprecation/deprecation
  targetChangeHandler: TargetChangeHandler; // eslint-disable-line deprecation/deprecation
}

/** Zone Targets React component.
 * @internal
 */
export class ZoneTargets extends React.Component<ZoneTargetsProps> {
  public override render(): React.ReactNode {
    switch (this.props.dropTarget) {
      case ZoneTargetType.Merge: // eslint-disable-line deprecation/deprecation
        return (
          <MergeTarget
            className={this.props.className}
            // eslint-disable-next-line deprecation/deprecation
            onTargetChanged={(isTargeted) => this.props.targetChangeHandler.handleTargetChanged(this.props.zoneId, ZoneTargetType.Merge, isTargeted)}
            style={this.props.style}
          />
        );
      case ZoneTargetType.Back: // eslint-disable-line deprecation/deprecation
        return (
          <BackTarget
            className={this.props.className}
            // eslint-disable-next-line deprecation/deprecation
            onTargetChanged={(isTargeted) => this.props.targetChangeHandler.handleTargetChanged(this.props.zoneId, ZoneTargetType.Back, isTargeted)}
            style={this.props.style}
            zoneIndex={this.props.zoneId}
          />
        );
      default:
        return null;
    }
  }
}
