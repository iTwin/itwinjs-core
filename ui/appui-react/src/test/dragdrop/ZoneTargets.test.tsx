/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
import { expect } from "chai";
import * as React from "react";
import * as sinon from "sinon";
import { WidgetZoneId, ZoneTargetType } from "@itwin/appui-layout-react";
import { ZoneTargets } from "../../appui-react/dragdrop/ZoneTargets";
import TestUtils, { userEvent } from "../TestUtils";
import { render } from "@testing-library/react";
/* eslint-disable deprecation/deprecation */
describe("ZoneTargets", () => {
  const spyMethod = sinon.spy();
  const handler = {
    handleTargetChanged: (_zoneId: WidgetZoneId, _type: ZoneTargetType, _isTargeted: boolean): void => {
      spyMethod();
    },
  };
  let theUserTo: ReturnType<typeof userEvent.setup>;
  beforeEach(()=>{
    theUserTo = userEvent.setup();
  });

  before(async () => {
    await TestUtils.initializeUiFramework();
  });

  after(() => {
    TestUtils.terminateUiFramework();
  });

  describe("DropTarget.Merge", () => {
    it("should call onTargetChanged", async () => {
      const {container} = render(<ZoneTargets zoneId={1} dropTarget={ZoneTargetType.Merge} targetChangeHandler={handler} />);

      await theUserTo.hover(container.querySelector(".nz-zone-6")!);
      expect(spyMethod.calledOnce).to.be.true;

      await theUserTo.unhover(container.querySelector(".nz-zone-6")!);
      expect(spyMethod.calledTwice).to.be.true;
    });
  });

  describe("DropTarget.Back", () => {
    it("should call onTargetChanged", async () => {
      spyMethod.resetHistory();
      const {container} = render(<ZoneTargets zoneId={1} dropTarget={ZoneTargetType.Back} targetChangeHandler={handler} />);

      await theUserTo.hover(container.querySelector(".nz-zone-1")!);
      expect(spyMethod.calledOnce).to.be.true;

      await theUserTo.unhover(container.querySelector(".nz-zone-1")!);
      expect(spyMethod.calledTwice).to.be.true;
    });
  });

});
