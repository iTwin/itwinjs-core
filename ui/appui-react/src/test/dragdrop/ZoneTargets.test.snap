// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneTargets DropTarget.Back renders correctly 1`] = `
<BackTarget
  onTargetChanged={[Function]}
  zoneIndex={1}
/>
`;

exports[`ZoneTargets DropTarget.Merge renders correctly 1`] = `
<MergeTarget
  onTargetChanged={[Function]}
/>
`;
